### PR TITLE
ccgx: fix wrong failure to replug

### PR DESCRIPTION
**Pull request by chlee75**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ x] Code fix
- [ ] Feature
- [ ] Documentation

Changes 
 1. Fix “failed to get device before update reload: ..." error happens sometimes when it waits for device replug.
  - Set "FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG" flag before sending a trigger code to the device.
  2. Fix  typo of function name.

### CODE REVIEW

File: plugins/ccgx/fu-ccgx-dmc-device.c [hunk 1]
```diff
@@ -616,17 +616,22 @@ fu_ccgx_dmc_device_attach(FuDevice *device, FuProgress *progress, GError **error
 	if (fu_device_get_update_state(self) != FWUPD_UPDATE_STATE_SUCCESS)
 		return TRUE;
 
+	if (manual_replug == FALSE)
+		fu_device_add_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
+
```
Comment (superm1) [inline plugins/ccgx/fu-ccgx-dmc-device.c:621]: I think it's better to move this as an else clause below by the other manual_replug check so it's more obvious what is different for manual replug

File: plugins/ccgx/fu-ccgx-dmc-device.c [hunk 2]
```diff
@@ -616,17 +616,22 @@ fu_ccgx_dmc_device_attach(FuDevice *device, FuProgress *progress, GError **error
...
 	if (self->update_model == DMC_UPDATE_MODEL_DOWNLOAD_TRIGGER) {
 		if (self->trigger_code > 0) {
 			if (!fu_ccgx_dmc_device_send_download_trigger(self,
 								      self->trigger_code,
 								      error)) {
+				fu_device_remove_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
```
Comment (superm1) [inline plugins/ccgx/fu-ccgx-dmc-device.c:627]: presumably this isn't needed if you move it to lower

Comment (chlee75) [reply plugins/ccgx/fu-ccgx-dmc-device.c:627]: Don't  I need remove FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG flag when return FALSE, if I set FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG flag before?

File: plugins/ccgx/fu-ccgx-dmc-device.c (hunk 1 above)
Comment (chlee75) [reply plugins/ccgx/fu-ccgx-dmc-device.c]: @superm1 
Could you check it at [0e64b65](https://github.com/fwupd/fwupd/pull/4563/commits/0e64b65ac32b78bca7ca98dbb0912f364db0b382) ?

File: plugins/ccgx/fu-ccgx-dmc-device.c [hunk 3]
```diff
@@ -631,14 +631,22 @@ fu_ccgx_dmc_device_attach(FuDevice *device, FuProgress *progress, GError **error
 			if (!fu_ccgx_dmc_device_send_download_trigger(self,
 								      self->trigger_code,
 								      error)) {
-				fu_device_remove_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
+				if (manual_replug)
```
Comment (hughsie) [inline plugins/ccgx/fu-ccgx-dmc-device.c:634]: can you add {} around this top-level block please.

File: plugins/ccgx/fu-ccgx-dmc-device.c [hunk 4]
```diff
@@ -631,14 +631,22 @@ fu_ccgx_dmc_device_attach(FuDevice *device, FuProgress *progress, GError **error
...
+					fu_device_remove_flag(device,
+							      FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
 				g_prefix_error(error, "download trigger error: ");
 				return FALSE;
 			}
 		}
 	} else if (self->update_model == DMC_UPDATE_MODEL_PENDING_RESET) {
 		if (!fu_ccgx_dmc_device_send_sort_reset(self, manual_replug, error)) {
-			fu_device_remove_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
+			if (manual_replug)
```
Comment (hughsie) [inline plugins/ccgx/fu-ccgx-dmc-device.c:646]: can you add {} around this top-level block please.

Comment (chlee75) [reply plugins/ccgx/fu-ccgx-dmc-device.c:646]: Do you mean below?
if (manual_replug) {
	fu_device_remove_flag(device, FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION);
} else {
	fu_device_remove_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);						  
}

Comment (hughsie) [reply plugins/ccgx/fu-ccgx-dmc-device.c:646]: yup, thanks.

Comment (chlee75) [reply plugins/ccgx/fu-ccgx-dmc-device.c]: Fix it at [4b2a452](https://github.com/fwupd/fwupd/pull/4563/commits/4b2a452ac98f068e95fe1677af0743c3aa20a2ab)

File: plugins/ccgx/fu-ccgx-dmc-device.c [hunk 5]
```diff
@@ -616,33 +616,44 @@ fu_ccgx_dmc_device_attach(FuDevice *device, FuProgress *progress, GError **error
...
 		if (self->trigger_code > 0) {
 			if (!fu_ccgx_dmc_device_send_download_trigger(self,
 								      self->trigger_code,
 								      error)) {
+				if (manual_replug) {
+					fu_device_remove_flag(device,
+							      FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION);
+				} else {
+					fu_device_remove_flag(device,
+							      FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
```
Comment (hughsie) [inline plugins/ccgx/fu-ccgx-dmc-device.c:639]: I understand removing FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG -- but do you really want to remove FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION too? It seems the hardware still needs activating, even if the reset failed, no? If you also want to remove FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION you also need to uninhibit  `update-pending` too.

Comment (chlee75) [reply plugins/ccgx/fu-ccgx-dmc-device.c:639]: FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION is set when device need manual re-plug such as Lenovo USB-C dock.
Is it OK not to remove "FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION" flag when return FALSE in attach?
If yes, I will not remove "FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION" flag.

Comment (chlee75) [reply plugins/ccgx/fu-ccgx-dmc-device.c]: @hughsie
FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION flag remain.
See [932298a](https://github.com/fwupd/fwupd/pull/4563/commits/932298a793b135c88c2db6bc866e35f2107f57cb)

**Pull request merged**